mirror-api/mirror-api.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+import argparse
+import boto3
+import datetime as dt
+import math
+import logging
+import os
+import requests
+import sys
+from urllib.parse import urlparse
+
+import asyncio
+import httpx
+from dateutil.parser import parse
+from pystac import Catalog, Collection, Item, CatalogType, layout, STAC_IO
+from pdb import set_trace
+
+logger = logging.getLogger(__name__)
+
+def s3_read(uri):
+    parsed = urlparse(uri)
+    if parsed.scheme == 's3':
+        bucket = parsed.netloc
+        key = parsed.path[1:]
+        s3 = boto3.resource('s3')
+        obj = s3.Object(bucket, key)
+        return obj.get()['Body'].read().decode('utf-8')
+    else:
+        return STAC_IO.default_read_text_method(uri)
```

Comment:
Don't know enough about this python code and the library. Just have a couple of questions. Why do we want to distinguish s3 and else? We only want to read/write from/to s3 right? where can we pass the bucket name to know which bucket to read/write from/to?

Reply:
The script can write to local storage or s3. Local catalog is useful for development.

The script takes in a path or s3 URL